pkgs/by-name/py/pyrefly/package.nix

```
@@ -0,0 +1,41 @@
+{
+  lib,
+  python3,
+  fetchPypi,
+  rustPlatform,
+  maturin,
+}:
+python3.pkgs.buildPythonApplication rec {
+  pname = "pyrefly";
+  version = "0.17.1";
+  format = "pyproject";
```

`format` is deprecated, actually 😅 

```suggestion
  pyproject = true;
```

---

I think `pyproject = true;` an indirect default nowadays. You can safely remove it. It causes a rebuild, and it has a different output. But after with `--check`:

```
error: derivation '/nix/store/pcinwg72q15sykpq7jc6pi8psh3sck2a-pyrefly-0.17.1.drv' may not be deterministic: output '/nix/store/a8w3n650dwvv0qiwly397wf7fq9m49dd-pyrefly-0.17.1-dist' differs
```

So it isn't even reproducible as of now. But `diffoscope with-pyproject-true without-pyproject-true` only reports differences in the python bytecode, the `pyrefly` binary, and `lib/python3.12/site-packages/pyrefly-0.17.1.dist-info/RECORD`, where the latter contains a sha256sum of the `pyrefly` binary so that doesn't really count.

Diffoscope: [diff.html.gz](https://github.com/user-attachments/files/20598248/diff.html.gz)

---

@dtomvan Thanks. Changed in https://github.com/NixOS/nixpkgs/commit/5c1f314ebb6ddee499d3efc5f25c14d2c8e160f1